src/openssl/ssl/socket.cr

```diff
@@ -181,4 +181,11 @@ abstract class OpenSSL::SSL::Socket < IO
   def unbuffered_rewind
     raise IO::Error.new("Can't rewind OpenSSL::SSL::Socket::Client")
   end
+
+  # This method allows to fetch the SSL SNI string if present
```

Comment:
Is this description correct? It's a bit weird that the method name is `hostname` but here it says "SSL SNI string".

Reply:
SNI or Server Name Indication is the used name for a hostname being passed inside the OpenSSL protocol.

I think saying `SNI` here makes more sense then hostname because we are already in the context of a Socket and I don't want mixups, if someone thinks this should change let me know

Reply:
Let's strip the "SSL", though. It's redundant in this context, let alone SSL as a protocol is pretty dead by now (everybody is using TLS). 

We could also say something like "Returns the hostname provided through Server Name Indication (SNI)"

Reply:
I like @jhass proposal, SSL is redundant and SNI may not be a known acronym.